nautobot/ui/src/components/Navbar.js

```diff
@@ -41,6 +45,35 @@ export function Navbar() {
     const location = useLocation();
     const { data: readyRoutes } = useGetNewUIReadyRoutesQuery();
 
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const changeHandler = (event) => {
+        let filters = [];
+        if (event.target.value !== "") {
+            filters = [["q", event.target.value]];
+        }
+        setSearchParams([
+            ...filters,
+            ...[...searchParams].filter(
+                ([searchParamLabel]) =>
+                    !FILTER_RESET_QUERY_PARAMS.includes(searchParamLabel) &&
+                    searchParamLabel !== "q"
```

Comment:
That expression would've looked nicer if `"q"` was a constant or part of a constant, as suggested in a previous comment.

Reply:
Actually, nevermind, after staring at it for a while, I prefer it the way you implemented it